web3/method.py

```diff
@@ -149,15 +153,29 @@ def __init__(
         self.is_property = is_property
 
     def __get__(
-        self, obj: Optional["Module"] = None, obj_type: Optional[Type["Module"]] = None
+        self,
+        module: Optional["Module"] = None,
+        _type: Optional[Type["Module"]] = None,
     ) -> TFunc:
-        if obj is None:
+        self._module = module
+        if module is None:
             raise Web3TypeError(
                 "Direct calls to methods are not supported. "
-                "Methods must be called from an module instance, "
+                "Methods must be called from a module instance, "
                 "usually attached to a web3 instance."
             )
-        return obj.retrieve_caller_fn(self)
+        if module.w3.provider._is_batching:
+            if self.json_rpc_method in RPC_METHODS_UNSUPPORTED_DURING_BATCH:
+                raise MethodNotSupported(
+                    f"Method `{self.json_rpc_method}` is not supported within a batch "
+                    "request."
+                )
+            return module.retrieve_request_information(self)
+        else:
+            return module.retrieve_caller_fn(self)
+
+    def __call__(self, *args: Any, **kwargs: Any) -> Any:
```

Comment:
Think about removing this?

Reply:
For clarity, this explicitly makes `Method` class callable... otherwise we get mypy conflicts, among other issues, such as:

```
web3/_utils/batching.py:128: error: "Method[Callable[..., Any]]" not callable  [operator]
```

The `batching.py` line in question:

```py
       for method, params in batch_payload.items():
            for param in params:
                self.add(method(param))  # <--- conflicting line
```